### PR TITLE
Provide a Debian package to automate deployment.

### LINKED ISSUE #31: How to handle changes to index schema?
Currently, re-indexing fails when the schema used by the indexer does not correspond to that stored on disk. We should determine how to handle this, e.g. by deleting or renaming the index directory.

### PR DESCRIPTION
**Pull request by adamreichold**
Closes #31

### CODE REVIEW

File: deployment/scripts/postinst [hunk 1]
```diff
@@ -0,0 +1,19 @@
+#!/bin/sh
+
+set -e
+
+DATA_PATH=/var/lib/umwelt-info
+
+if [ -e "$DATA_PATH/index" ]
```
Comment (adamreichold) [inline deployment/scripts/postinst]: This is not really a complete solution as it does not create the `umwelt-info` user nor the `/var/lib/umwelt-info` directory owned by it.

Maybe we should create both via

```console
> useradd --home-dir /var/lib/umwelt-info --create-home umwelt-info
```

if the directory is missing? And then create (and chown it to `umwelt-info:umwelt-info`) an empty `datasets` folder so the indexer can run and the server can start properly?

Comment (adamreichold) [reply deployment/scripts/postinst]: Added that. Will need to test this tomorrow by removing the existing installation...

Comment (adamreichold) [reply deployment/scripts/postinst]: Hopefully found a much simpler solution noticing that the indexer will always create an index even if reading the datasets fails. This means the server will come up with a missing datasets (and thereby an empty index) even though it wont serve anything until the harvester has run.

File: deployment/postinst [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+USER=umwelt-info
+DATA_PATH=/var/lib/umwelt-info
+
+if [ ! -e $DATA_PATH ]; then
+  useradd --home-dir $DATA_PATH --create-home $USER
+else
+  mv $DATA_PATH/index $DATA_PATH/index.old
+fi
+
+systemctl start indexer.service
```
Comment (jakobdeller) [inline deployment/postinst:12]: What happens if at this point the indexer is for some reason running already? Like doing two installs in a short time period?

Comment (adamreichold) [reply deployment/postinst:12]: This will block and run the indexer again after is has finished.

**Review by jakobdeller — APPROVED** [review]

**Pull request merged**